Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -393,8 +393,20 @@ public T CreateInstance<T>()
         /// <returns>System.Object.</returns>
         protected object CreateInstanceSafe(Type type)
         {
+            if (_creatingInstances == null)
+            {
+                _creatingInstances = new List<Type>();
+            }
+
+            if (_creatingInstances.IndexOf(type) != -1)
+            {
+                Logger.LogError("Incompatible plugin detected. Attempting to bypass.");
+                throw new ExternalException("Incompatible plugin.");
```

Comment:
Doesn't have to be a plugin and also the function is called `CreateInstanceSafe`. It's no longer safe ;)

Reply:
Now called CreateInstanceWithSafeguards

Naff name, but i couldn't think of better name besides CreateInstanceInASlightlyMoreSecureWayButItStillMightFailAndDontPutItInAnExceptionHandler.